app/backend/routes/user/types.ts

```diff
@@ -0,0 +1,45 @@
+import { z } from "zod";
+import type { ServerWebSocket } from "bun";
+import type { Env as HonoEnv } from "hono";
+import type { Session, User } from "#db/schema.sql";
+
+// * Set of rules for the profile
+// * Defines how the data should be validated
+// * Set what are valid inputs and what fields are allowed
+
+export type ChatSocket = ServerWebSocket<{ user: string }>;
+
+export interface Env extends HonoEnv {
+	Variables: {
+		user: User | null;
+		session: Session | null;
+	};
+}
```

Comment:
No need to redefine these

Reply:
Yes, the interface is redundant code, but I created it to make the source of user and session more obvious. But I can remove it if u want.